### PR TITLE
Add cram support + read splitting with seqkit for speedup

### PR DESCRIPTION
**Pull request by FriederikeHanssen**
crams-everywhere_meme

BAMQC is still a bit of an open problem, especially after base recalibration....

Important change: BamQC + Samtools Stats is only done ONCE BEFORE BaseRecalibration right now: if duplicates are marked it is run after duplicate marking, if they are not it is run after mapping. This reduces the runtime, since MarkDuplicates can take care of merging split reads internally without runtime punishments. 
Spark implementation currently only works with singularity. The docker image has issue, and would have to possibly be rebuild :( 

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
    - [ ] If you've added a new tool - add to the software_versions process and a regex to `scrape_software_versions.py`
    - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)
    - [ ] If necessary, also make a PR on the nf-core/sarek _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint .`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by maxulysse** [comment]
will close #63

**Comment by maxulysse** [comment]
@FriederikeHanssen your branch is out-of-date with the base branch due to #390
You might want to sync

**Comment by FriederikeHanssen** [comment]
ups, thanks for the heads up

**Comment by FriederikeHanssen** [comment]
@maxulysse  I cleaned up the code a bit now. there are still quite some things that need to be fixed/discussed in separate PRs. I'll add a collection here for context and turn it into cards on the project board: 

Things that need to still be fixed from PR #388

- [ ] Adding CSV writing after mapping back in
- [ ] Adding CSV wrirting after duplicate marking back in
- [ ] allow results to be output in bam format
- [ ] Warning if patient/sample are not part of samplesheet
- [ ] Add Merging step of bam files after mapping, if mapping output should be saved
- [ ] fix the channel/file magic that is going on with dbsnp/known indels
- [ ] discuss if we really should have BQSR Spark still in Beta mode. If we keep it, label in big bold letters that it is very experimental
- [ ] better naming for duplicatemarking workflow. Separating it out is tricky in my eyes. But now we are covering two functionalities
- [ ] if skip_markduplicates than map directly to cram and not to bam (see next point too)
- [ ] simplify MDSpark and use cram bamqc container both times. code would be less complicated. Depends on how much longer bamqc would take
- [ ] running multiple tools fails sometimes
- [ ] DeepVariant to cram, modules.conf not passed down

The CI tests are not passing anymore, since they are not bams but crams. I don't really know what is going on with the nf-core lint  testing. the error message is a bit cryptic to me.

File: conf/modules.config [hunk 1]
```diff
@@ -17,7 +17,7 @@ params {
         }
         'bwamem2_index' {
             publish_dir      = 'reference'
-            publish_files    = false
+            publish_files    = true
```
Comment (maxulysse) [inline conf/modules.config:20]: I'd assume we want it to be true only when saving the reference

File: modules/local/index_target_bed/main.nf [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    output:
+    tuple path("${target_bed}.gz"), path("${target_bed}.gz.tbi")
+
+    script:
+    """
+    bgzip --threads ${task.cpus} -c ${target_bed} > ${target_bed}.gz
+    tabix ${target_bed}.gz
+    """
+}
```
Comment (maxulysse) [inline modules/local/index_target_bed/main.nf:33]: I added that in nf-core/modules

File: nextflow.config [hunk 3]
```diff
@@ -112,10 +111,18 @@ params {
 }
 
 // Load base.config by default for all pipelines
-includeConfig 'conf/base.config'
+try {
+  includeConfig 'conf/base.config'
+} catch (Exception e) {
+  System.err.println("WARNING: Could not load nf-core/config profiles: ${params.custom_config_base}/base.config")
```
Comment (maxulysse) [inline nextflow.config:117]: Note sure about that

File: nextflow.config [hunk 4]
```diff
@@ -112,10 +111,18 @@ params {
...
+}
 
 // Load modules.config by default for all pipelines
-includeConfig 'conf/modules.config'
+try {
+  includeConfig 'conf/modules.config'
+} catch (Exception e) {
+  System.err.println("WARNING: Could not load nf-core/config profiles: ${params.custom_config_base}/modules.config")
```
Comment (maxulysse) [inline nextflow.config:124]: same here

**Review by maxulysse — COMMENTED** [review]
Love it

File: nextflow.config (hunk 3 above)
Comment (FriederikeHanssen) [reply nextflow.config:117]: ok, yeah I had problems were the error messsage for failing to load configs was very confusing. But maybe this is also something to deal with upstream and not in sarek

Comment (maxulysse) [reply nextflow.config:117]: no worry, we'll fix that

**Pull request merged**